pkg/volume/emptydir/quota.go

```diff
@@ -123,14 +138,19 @@ func (xqa *xfsQuotaApplicator) Apply(dir string, medium api.StorageMedium, pod *
 
 func (xqa *xfsQuotaApplicator) applyQuota(volDevice string, quota resource.Quantity, fsGroupID int64) error {
 	_, stderr, err := xqa.cmdRunner.RunApplyQuotaCommand(volDevice, quota, fsGroupID)
-	if err != nil {
-		return err
-	}
+
 	// xfs_quota is very happy to fail but return a success code, likely due to its
 	// interactive shell approach. Grab stderr, if we see anything written to it we'll
 	// consider this an error.
+	//
+	// If we exit non-zero *and* write to stderr, stderr is likely to have the details on what
+	// actually went wrong, so we'll use this as the error message instead.
 	if len(stderr) > 0 {
-		return fmt.Errorf("xfs_quota wrote to stderr: %s", stderr)
+		return fmt.Errorf("error applying quota: %s", stderr)
+	}
+
+	if err != nil {
```

Comment:
err being checked first was what actually bit me when fixing this issue, you get a generic "system exit 1" error in the logs when your real info went to stderr and isn't visible.

xfs_quota behaves a little weird, it's more of an interactive CLI shell but can also be called the way we do to one-shot things. During original development I saw return 0 at times even though the internal command you asked it to run failed in some situations. In these situations, as well as the one this PR fixed, the real info was on stderr. I'm not aware of any situation where err will contain anything other than system exit 1, so as far as I know we're better off with it in this order.